File: backend/app/src/models/Player.py
from .locations.HomePlanet import HomePlanet
from .locations.Planet import Planet



class Player():
    def __init__(self, name: str, player_id: int, bases: list = None):
        if bases is None:
            bases = [HomePlanet("Heimatplanet")]
        self.__action_list: list[dict] =  [
                                            { "name": "Select Base", "action": self.select_base, }
                                        ]
        self.name: str = name
        self.player_id: int = player_id
        self.__bases: list = bases

    def get_bases(self) -> list:
        return self.__bases
    
    def fetch_player_state(self) -> dict:
        return {
            "category": "player",
            "name": self.name,
            "player_id": str(self.player_id),
            "base_states": [base.fetch_base_state() for base in self.__bases]
        }



    def match_payload_action(self, action: str, context: list) -> dict:
        try:
            for act in self.__action_list:
                if act.get("name") == action:
                    return act["action"](*context)
                
        except Exception as e:
            return  { "success": False, "message": f"{e}" }


    def select_base(self, target) -> dict:
        for base in self.__bases:
            if base.get_base_id() == int(target):
                return  { "success": True, "target": base }
        
        return  { "success": False, "message": f"{target} nicht gefunden!", "target": f"{target} nicht gefunden!" }

File: backend_refactoring/CSharpApi/Models/GamePlayObjects/Player.cs
using CSharpApi.Models.AstronomicalObjects;
using CSharpApi.Models.BluePrints.Locations;
using Microsoft.EntityFrameworkCore;
using Microsoft.OpenApi.Any;
using System;
using System.Collections;
using System.Collections.Generic;

namespace CSharpApi.Models.GamePlayObjects
{
    public class Player
    {
        private Guid Id;
        private string Name { get; set; } = string.Empty;

        private List<Base> Bases { get; set; } = [];

        private List<Dictionary<string, object>> Action { get; set; }


        public Player(string name, List<Base> bases)
        {
            Id = Guid.CreateVersion7(DateTime.UtcNow);

            Name = name;

            if (bases != null)
            {
                Bases = bases;
            }
            else
            {
                Bases.Add(new HomePlanet("Heimatplanet", []));
            }
        }

        public Dictionary<string, object> FetchPlayerState()
        {
            List<object> baseState = new();

            foreach (var bases in Bases)
            {
                baseState.Add(bases.FetchBaseState());

            }

            return new Dictionary<string, object> {
            {"category", "player" },
            {"name", Name },
            {"player_id", Id },
            {"base_states", baseState } };
        }

        public Dictionary<string, object> MatchPayloadAction(string action, List<object> context)
        {
            foreach (Dictionary<string, object> item in Action)
            {
                string lookup = item.GetValueOrDefault("name", "").ToString()!;

                if (!string.IsNullOrEmpty(lookup))
                {
                    lookup = lookup.ToString()!;
                    return new Dictionary<string, object>
                    {
                        { "action", lookup },
                        { "context", context }

                    };
                }
            }

            return new Dictionary<string, object> {
                    {"success", false},
                    {"message", $"No value for Key: NAME!"}
            };
        }

        public Dictionary<string, object> SelectBase(int target)
        {
            foreach (var targetBase in Bases)
            {
                if (targetBase.BaseId == target)
                {
                    return new Dictionary<string, object> {
                        {"success", true},
                        {"target", targetBase}
                    };
                }
            }

            return new Dictionary<string, object> {
                { "success", false },
                { "message", $"{target} nicht gefunden!" },
                { "target", $"{target} nicht gefunden!" }
            };
        }

    }
}